### ISSUE TITLE
[a11y]: Keyboard operation for menu on pagination component

### ISSUE THREAD

**Issue by thefirstartist** [carbon-design-system/carbon#14686]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

v11

### React version

v1.38

### Automated testing tool and ruleset

n/a

### Description

1. It seems the dropdowns are opened with Space or with up or down arrows only. However, ARIA APG suggests using Enter key as the default key and the Space as optional to open the menu. ([ARIA Guidance](https://www.w3.org/WAI/ARIA/apg/patterns/menubar/)). We also use Enter key to open the menu in Carbon's [Dropdown](https://carbondesignsystem.com/components/dropdown/usage/) component. Shouldn't we do the same here to be consistent?
2. When the menu is opened, pressing Tab key should be able to close the menu and move the focus to the next focusable item.
3. A pagination contains multiple menu items and navigation buttons. It is a composite widgets as described in the practice for [Keyboard Navigation Inside Components](https://www.w3.org/WAI/ARIA/apg/practices/keyboard-interface/#kbd_general_within), similar to a [menubar](https://www.w3.org/WAI/ARIA/apg/patterns/menubar/examples/menubar-editor/) or [toolbar](https://www.w3.org/WAI/ARIA/apg/patterns/toolbar/examples/toolbar/). I think it makes more sense to move the focus with arrow keys, instead of the tab key, inside the component. 

### WCAG 2.1 Violation

WCAG 2.2.1 Keyboard

### Reproduction/example

n/a

### Steps to reproduce

[Component live demo](<img width="853" alt="image" src="https://github.com/carbon-design-system/carbon/assets/40870771/4f15776d-9fd5-42bb-adc6-498139a94e3b">)
<img width="853" alt="image" src="https://github.com/carbon-design-system/carbon/assets/40870771/2240c1b9-50fe-4334-a43b-7ec3d1484846">

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by mbgower** [comment]
What you folks are encountering is the native HTML `select` component by default on a Mac. While I agree with Sam that it would be nice if it behaved like a dropdown and opened on Enter, the fact is that it does not.
Note that if you are on a Windows system, you can Tab away from an open select. But the same limitation that it opens only with Space exists on the Windows system. It also won't open with down arrow on either OS (it simply cycles through the choices without opening).

So this isn't a bug. It is expected behaviour.

In regard to the request to make arrow keys move between components, it would be technically very difficult to do here, and I don't see it as necessary or even better. The values in a Select are cycled with both the up/down arrow keys AND the left/right arrow keys. So how would you be able to move away from a select with an arrow key? The normal way of moving away from an input is Tab, and that works fine here.

**Comment by thefirstartist** [comment]
Yes, it would be nice if it behaved like a dropdown and expand on Enter. It's an expected behavior for a dropdown, accordion, and most elements with a chevron icon. I understand the native HTML select component does not support Enter on a Mac, but it looks like a defect from users' perspective. Enter key expands and collapses dropdowns on Gmail, Yahoo mail, Google, and it also works on Oracle, Salesforce, and pretty much everywhere I familiar with.
<img width="647" alt="image" src="https://github.com/carbon-design-system/carbon/assets/40870771/e87dff63-6324-4d11-afc4-54a3723deb55">

**Comment by tw15egan** [comment]
We purposely do not override any of the browser defaults for the `Select` component.  All of these functionalities should exist in the `Dropdown` component, so outside of the `Pagination` component, you can use that. 

I believe the reason we use `Select` in the `Pagination` component is because it is better suited to handle large lists of numbers (we've gotten issues that people have hundreds of items in their `Pagination` select), and will overlay any containers with `overflow: hidden` it is contained inside. Once the `floating-ui` improvements have been made (https://github.com/carbon-design-system/carbon/issues/14139), we can take a look and see if it is possible to refactor the `Pagination` to use `Dropdown`. For right now, however, we don't plan on making any changes to the behavior.
